clippy_lints/src/missing_inline.rs

```diff
@@ -121,20 +124,7 @@ impl<'tcx> LateLintPass<'tcx> for MissingInline {
                     }
                 }
             },
-            hir::ItemKind::Const(..)
-            | hir::ItemKind::Enum(..)
-            | hir::ItemKind::Macro(..)
-            | hir::ItemKind::Mod(..)
-            | hir::ItemKind::Static(..)
-            | hir::ItemKind::Struct(..)
-            | hir::ItemKind::TraitAlias(..)
-            | hir::ItemKind::GlobalAsm { .. }
-            | hir::ItemKind::TyAlias(..)
-            | hir::ItemKind::Union(..)
-            | hir::ItemKind::ExternCrate(..)
-            | hir::ItemKind::ForeignMod { .. }
-            | hir::ItemKind::Impl { .. }
-            | hir::ItemKind::Use(..) => {},
+            _ => {},
```

Comment:
Is there a reason for this change? I'd prefer an exhaustive match here so that we get a proper error here and are reminded to potentially handle new items that could need `#[inline]` whenever a new one is added